controllers/provisioning_controller.go

```diff
@@ -223,48 +219,32 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	}
 
 	// If Metal3 Deployment already exists, do nothing.
-	exists, err := provisioning.GetExistingMetal3Deployment(r.KubeClient.AppsV1(), ComponentNamespace)
+	existing, err := provisioning.GetExistingMetal3Deployment(r.KubeClient.AppsV1(), ComponentNamespace)
 	if err != nil && !apierrors.IsNotFound(err) {
 		return ctrl.Result{}, errors.Wrap(err, "failed to find existing Metal3 Deployment")
 	}
 
-	if exists {
+	if existing != nil {
 		r.Log.V(1).Info("metal3 deployment already exists")
-		err = r.updateCOStatus(ReasonComplete, "found existing Metal3 deployment", "")
+		provisioning.AnnotateExistingMetal3Deployment(existing)
```

Comment:
For right now we are assuming the deployment we would have created won't be any different from the one the machine-api-operator creates. That assumption will not hold for very long, though. I think we want to unify all of this logic to not treat the 2 cases quite so differently. Even if we find an existing deployment, we should make sure it matches the one we would have created, including the annotation. That will also let us reset the definition of the deployment if someone tries to edit it later, for example.

We should also say that we are Progressing until the Deployment reports that it is Available. That way we don't indicate success too soon, and leave an issue with the deployment undiscovered.

Reply:
Agree with point #1. I hadn't considered that while during implementation.

Regarding putting the CO in progressing state, not sure how I missed that!

Reply:
Both the above comments are addressed now.